### PR TITLE
Feature/v75 update jenkinsfile

### PR DESCRIPTION
**Pull request by cookersjs**
- JF updated to use shared groovy library
- Fairly complex implementation that builds an intermediate BioModels graph DB (that requires interactions) and analysis core before running the Perl and Java BioModels programs.

### CODE REVIEW

**Review by SolomonShorser-OICR — DISMISSED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -1,170 +1,163 @@
...
+def utils = new Utilities()
+
 pipeline {
 	agent any
 
 	stages {
-  		// This stage checks that an upstream project, AddLinks-Insertion, was run successfully for its last build.
-		stage('Check OrthoinferenceStableIdentifierHistory build succeeded'){
+  		// This stage checks that two upstream projects, OrthoinferenceStableIdentifierHistory and AddLinks-Insertion, were run successfully for their last build.
+		stage('Check OrthoinferenceStableIdentifierHistory  and AddLinks-Insertion builds succeeded'){
```
Comment (jweiser) [inline Jenkinsfile]: Nit pick :) - double space just before "and"?

Comment (SolomonShorser-OICR) [reply Jenkinsfile]: :roll_eyes:

File: Jenkinsfile [hunk 2]
```diff
@@ -1,170 +1,163 @@
...
-		}
+		// Runs the perl 'biomodels.pl' script that builds the models2pathways.tsv file from release_current DB.
 		stage('Main: Generate BioModels file'){
 			steps{
 				script{
+					def releaseVersion = utils.getReleaseVersion()
+					
 					dir("${env.ABS_RELEASE_PATH}/biomodels/"){
+						// Downloads the BioModels_Database-r31_pub-sbml_files that are stored on S3.
+						def biomodelsDatabaseArchive = "BioModels_Database-r31_pub-sbml_files"
```
Comment (jweiser) [inline Jenkinsfile:96]: I believe this file changes and is updated by BioModels to use a new release number.  Is there a reason why we're using r31 here?

Comment (SolomonShorser-OICR) [reply Jenkinsfile:96]: Good eye!

File: Jenkinsfile [hunk 3]
```diff
@@ -1,170 +1,163 @@
...
+						sh "aws s3 --no-progress cp --recursive ${env.S3_RELEASE_DIRECTORY_URL}/supplementary_files/${biomodelsDatabaseArchive}.tar.bz2 ."
+						// Runs perl script and then moves the models2pathways.tsv file to the downloads directory.
 						withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
-							sh "perl biomodels.pl -db ${env.RELEASE_CURRENT}"
-							sh "cp models2pathways.tsv ${env.ABS_DOWNLOAD_PATH}/${currentRelease}/"
+							sh "perl biomodels.pl -db ${env.RELEASE_CURRENT_DB}"
```
Comment (jweiser) [inline Jenkinsfile:100]: We should look into integrating the logic from this script into the Java BioModels release step.

File: Jenkinsfile (hunk 2 above)
Comment (cookersjs) [reply Jenkinsfile:96]: I didn't know how the file was used - I'll look into this

File: Jenkinsfile (hunk 3 above)
Comment (cookersjs) [reply Jenkinsfile:100]: Agreed! I'll look into it to see if it looks complex - not sure why it hasn't been incorporated yet

File: Jenkinsfile (hunk 1 above)
Comment (cookersjs) [reply Jenkinsfile]: Nice catch

File: Jenkinsfile (hunk 2 above)
Comment (cookersjs) [reply Jenkinsfile:96]: I've made a note of this and will see about correcting it for the dry release run

**Review by jweiser — APPROVED** [review]

**Pull request merged**